helix_fhir_client_sdk/graph/simulated_graph_processor_mixin.py

```diff
@@ -243,79 +243,83 @@ async def _process_target_async(
-                # iterate through all references
-                if parent_resource and references and target_type:
-                    reference_ids: List[str] = [
-                        cast(str, r.get("reference"))
-                        for r in references
-                        if "reference" in r and isinstance(r, dict)
-                    ]
-                    for reference_id in reference_ids:
-                        reference_parts = reference_id.split("/")
-                        if reference_parts[0] == target_type:
-                            child_id = reference_parts[1]
-                            (
-                                child_response,
-                                cache_hits,
-                            ) = await self._get_resources_by_parameters_async(
-                                session=session,
-                                resource_type=target_type,
-                                id_=child_id,
-                                cache=cache,
-                                scope_parser=scope_parser,
-                                logger=logger,
-                            )
-                            responses.append(child_response)
-                            children = child_response.get_bundle_entries()
-                            if logger:
-                                logger.info(
-                                    f"Received child resources"
-                                    + f" from parent {parent_resource.get('resourceType')}/{parent_resource.get('id')}"
-                                    + f" [{path}]"
-                                    + f", count:{len(child_response.get_resource_type_and_ids())}, cached:{cache_hits}"
-                                    + f", {','.join(child_response.get_resource_type_and_ids())}"
-                                )
-            else:  # single reference
-                if parent_resource and parent_resource.get(path) and target_type:
-                    reference = parent_resource.get(path)
-                    if reference and "reference" in reference:
-                        reference_id = reference["reference"]
-                        reference_parts = reference_id.split("/")
-                        if reference_parts[0] == target_type:
-                            child_id = reference_parts[1]
-                            (
-                                child_response,
-                                cache_hits,
-                            ) = await self._get_resources_by_parameters_async(
-                                session=session,
-                                resource_type=target_type,
-                                id_=child_id,
-                                cache=cache,
-                                scope_parser=scope_parser,
-                                logger=logger,
+
+        # forward link and iterate over list
+        if path and "[x]" in path:
```

Comment:
need to check if it is in the path instead of just the end, in order to support values both `foo.bar[x]` and `foo[x].bar`